crates/polars-lazy/src/frame/mod.rs

```diff
@@ -2160,6 +2162,43 @@ impl JoinBuilder {
             opt_state |= OptFlags::FILE_CACHING;
         }
 
+        // Decompose `is_between` predicates to allow for cleaner expression of range joins
+        #[cfg(feature = "is_between")]
+        let predicates: Vec<Expr> = predicates
```

Comment:
nit: instead of allocating multiple vecs in the iterator, we can pre-allocate 1 vec up front and push to that one.

Reply:
<facepalm />
will tweak accordingly, oops 😂

Reply:
Done